cognite/client/_api_client.py

```diff
@@ -609,6 +628,122 @@ def _aggregate(
         res = self._post(url_path=resource_path + "/aggregate", json=body, headers=headers)
         return [cls(**agg) for agg in res.json()["items"]]
 
+    @overload
+    def _advanced_aggregate(
+        self,
+        aggregate: Literal["count", "cardinalityValues", "cardinalityProperties"],
+        properties: EnumProperty
+        | str
+        | list[str]
+        | tuple[EnumProperty | str | list[str], AggregationFilter]
+        | None = None,
+        path: EnumProperty | str | list[str] | None = None,
+        query: str | None = None,
+        filter: CogniteFilter | dict | None = None,
+        advanced_filter: Filter | dict | None = None,
+        aggregate_filter: AggregationFilter | dict | None = None,
+        limit: int | None = None,
+    ) -> int:
+        ...
+
+    @overload
+    def _advanced_aggregate(
+        self,
+        aggregate: Literal["uniqueValues", "uniqueProperties"],
+        properties: EnumProperty
+        | str
+        | list[str]
+        | tuple[EnumProperty | str | list[str], AggregationFilter]
+        | None = None,
+        path: EnumProperty | str | list[str] | None = None,
+        query: str | None = None,
+        filter: CogniteFilter | dict | None = None,
+        advanced_filter: Filter | dict | None = None,
+        aggregate_filter: AggregationFilter | dict | None = None,
+        limit: int | None = None,
+    ) -> UniqueResultList:
+        ...
+
+    def _advanced_aggregate(
```

Comment:
This should have some tests in `test_api_client` (either unit or integration) - if iterating on this in the future we don't want to have to add duplicate tests on all the individual resources. Great that we have integration tests there aswell, but we want one authoritative place in the tests for the generic functionality.

Reply:
Not much logic to test, but added a few tests now.